Code/GraphMol/ChemTransforms/catch_tests.cpp

```diff
@@ -56,3 +58,202 @@ TEST_CASE("Github #1039", "[]") {
+        for(unsigned int i=0;i<m->getNumBonds();++i) {
+            for(unsigned int j=0;j<m->getNumBonds();++j) {
+                if(i!=j) {
+                    std::vector<unsigned int> bonds{i,j};
+                    auto resa = RDKit::MolFragmenter::fragmentOnBonds(*m, bonds);
+                    MolzipParams p;
+                    p.label = MolzipLabel::FragmentOnBonds;
+                     CHECK(MolToSmiles(*molzip(*resa,p)) == MolToSmiles(*m));
+                    
+                    // Now try using atom labels
+                    auto res = RDKit::MolFragmenter::fragmentOnBonds(*m, bonds, true, &dummyLabels);
+                    for(auto *atom : res->atoms()) {
+                        if(atom->getIsotope()) {
+                            atom->setAtomMapNum(atom->getIsotope());
+                        }
+                    }
+                    CHECK(MolToSmiles(*molzip(*res)) == MolToSmiles(*m));
+                }
+            }
+        }
+    }
+    
+    SECTION("test bond stereo")
+    {
+       auto a = "F/C=C/[*:1]"_smiles;
+       auto b = "[*:1]F"_smiles;
+       auto mol = molzip(*a,*b);
+      CHECK(MolToSmiles(*mol) == "F/C=C/F");
+    }
+    {
+        auto b = "F/C=C/[*:1]"_smiles;
+        auto a = "[*:1]F"_smiles;
+        auto mol = molzip(*a,*b);
+       CHECK(MolToSmiles(*mol) == "F/C=C/F");
+     }
+    
+    {
+          auto a = "O/C=N/[*:1]"_smiles;
+          auto b = "[*:1]C=C"_smiles;
+          auto mol = molzip(*a,*b);
+         CHECK(MolToSmiles(*mol) == "C=C/N=C/O");
+       }
+    {
+       auto b = "O/C=N/[*:1]"_smiles;
+       auto a = "[*:1]C=C"_smiles;
+       auto mol = molzip(*a,*b);
+      CHECK(MolToSmiles(*mol) == "C=C/N=C/O");
+    }
+    
+    {
+        auto a = "C=C/N=C/[*:1]"_smiles;
+        auto b = "O[*:1]"_smiles;
+        auto mol = molzip(*a,*b);
+        CHECK(MolToSmiles(*mol) == "C=C/N=C/O");
+    }
+    
+    {
+        auto b = "C=C/N=C/[*:1]"_smiles;
+        auto a = "O[*:1]"_smiles;
+        auto mol = molzip(*a,*b);
+        CHECK(MolToSmiles(*mol) == "C=C/N=C/O");
+    }
+    {
+        auto a = "C=C[*:1]"_smiles;
+        auto b = "O/C=N/[*:1]"_smiles;
+        auto mol = molzip(*a,*b);
+        CHECK(MolToSmiles(*mol) == "C=C/N=C/O");
+    }
+    {
+        auto b = "C=C[*:1]"_smiles;
+        auto a = "O/C=N/[*:1]"_smiles;
+        auto mol = molzip(*a,*b);
+        CHECK(MolToSmiles(*mol) == "C=C/N=C/O");
+    }
+    
+    {
+        auto a = "C=C[*:1].O/C=N/[*:1]"_smiles;
+        auto mol = molzip(*a);
+        CHECK(MolToSmiles(*mol) == "C=C/N=C/O");
+    }
+    {
+        auto a = "C=C[1*:1].O/C=N/[1*:1]"_smiles;
+        auto mol = molzip(*a);
+        CHECK(MolToSmiles(*mol) == "C=C/N=C/O");
+    }
+    /* For some reason, the molzip and fragment on bonds is having issues with bond stereo in some cases
+     roundtripping through moltosmiles "fixes" it though */
+    {
```

Comment:
I think this set of tests needs to be cleaned up a bit now that #3649 has been merged?

Reply:
I had just left the comment accidentally, I don't round trip through smiles anymore.